streamer/src/nonblocking/quic.rs

```diff
@@ -186,35 +187,55 @@ async fn handle_connection(
         stats.total_connections.load(Ordering::Relaxed),
     );
     while !stream_exit.load(Ordering::Relaxed) {
-        match uni_streams.next().await {
-            Some(stream_result) => match stream_result {
-                Ok(mut stream) => {
-                    stats.total_streams.fetch_add(1, Ordering::Relaxed);
-                    stats.total_new_streams.fetch_add(1, Ordering::Relaxed);
-                    let mut maybe_batch = None;
-                    while !stream_exit.load(Ordering::Relaxed) {
-                        if handle_chunk(
-                            &stream.read_chunk(PACKET_DATA_SIZE, false).await,
-                            &mut maybe_batch,
-                            &remote_addr,
-                            &packet_sender,
-                            stats.clone(),
-                            stake,
-                        ) {
-                            last_update.store(timing::timestamp(), Ordering::Relaxed);
-                            break;
+        if let Ok(stream) = tokio::time::timeout(
+            Duration::from_millis(WAIT_FOR_STREAM_TIMEOUT_MS),
+            uni_streams.next(),
+        )
```

Comment:
An alternative approach would be to make `stream_exit` a channel so that we could use it as a signal that could interrupt waiting for the `uni_streams.next()` future or the `read_chunk` future below.

Reply:
Yes, channel will be nicer to have.

Reply:
Could you please create a follow up PR to switch to a signal based approach?

Reply:
Can we just explicitly close the connection? that will exit all streams: https://github.com/quinn-rs/quinn/issues/1376#issuecomment-1163722291